tw-tasks-core/src/main/java/com/transferwise/tasks/triggering/KafkaTasksExecutionTriggerer.java

```diff
@@ -170,24 +168,29 @@ public void trigger(BaseTask task) {
     // TODO: Future improvement: try to also trigger in the same node, if there is room or more specifically if it is idle (for min latency)
     // TODO: Maybe needs another concurrency control for that. E.g. only trigger in node, when conc < 5, even max conc is 10.
 
+    // We need to give a non-null random key, so triggers will be evenly spread around partitions.
+    // With <null> key, the kafka client will start doing some kind of batch partitioning.
+
+    String key = String.valueOf((char) ThreadLocalRandom.current().nextInt(0xFFFF));
```

Comment:
`UuidUtils.generatePrefixCombUuid().toString()` was too expensive?

Reply:
Relatively to the new solution yes, but it did not matter too much on grand scale. Just a minor tweak.